### PR TITLE
Make pagination an option for FieldFunc

### PR DESCRIPTION
**Pull request by berfarah**
Don't require PaginateFieldFunc for creating field funcs. Rather, FieldFunc takes in a Paginate option.

### CODE REVIEW

File: graphql/schemabuilder/types.go [hunk 1]
```diff
@@ -21,11 +21,18 @@ type paginationObject struct {
 
 // FieldFuncOption is an interface for the variadic options that can be passed
 // to a FieldFunc for configuring options on that function.
-type FieldFuncOption func(*method)
+type FieldFuncOption interface {
+	option(*method)
```
Comment (willhug) [inline graphql/schemabuilder/types.go]: maybe `apply`?

File: graphql/schemabuilder/types.go [hunk 2]
```diff
@@ -85,6 +90,9 @@ func (s *Object) Key(f string) {
 type method struct {
 	MarkedNonNullable bool
 	Fn                interface{}
+
+	// Connection configuration
```
Comment (willhug) [inline graphql/schemabuilder/types.go]: ?  Is this meant as a header for the section?

File: graphql/schemabuilder/types.go (hunk 1 above)
Comment (berfarah) [reply graphql/schemabuilder/types.go]: Done

File: graphql/schemabuilder/types.go (hunk 2 above)
Comment (berfarah) [reply graphql/schemabuilder/types.go]: Yeah - I can remove it for now if it's confusing

File: graphql/schemabuilder/types.go [hunk 3]
```diff
@@ -21,14 +20,27 @@ type paginationObject struct {
...
+
+// fieldFuncOptionFunc is a helper to define FieldFuncOptions from a func.
+type fieldFuncOptionFunc func(*method)
+
+func (f fieldFuncOptionFunc) option(m *method) { f(m) }
 
 // NonNullable is an option that can be passed to a FieldFunc to indicate that
 // its return value is required, even if the return value is a pointer type.
-func NonNullable(m *method) {
+var NonNullable fieldFuncOptionFunc = func(m *method) {
```
[the remark this answers is not in the document]
Comment (stephen) [reply graphql/schemabuilder/types.go]: too cute

File: graphql/schemabuilder/pagination.go [hunk 4]
```diff
@@ -335,12 +335,10 @@ func getConnection(key string, out []reflect.Value, args PaginationArgs, returns
...
 // to be returned in the resolver func.
+//
+// Deprecated: Use FieldFunc(name, func, Paginated) instead.
 func (o *Object) PaginateFieldFunc(name string, f interface{}) {
-	o.paginatedFields = append(o.paginatedFields,
-		paginationObject{
-			Name: name,
-			Fn:   f,
-		})
+	o.FieldFunc(name, f, Paginated)
```
[the remark this answers is not in the document]
Comment (stephen) [reply graphql/schemabuilder/pagination.go]: love it

File: graphql/schemabuilder/types.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (stephen) [reply graphql/schemabuilder/types.go]: maybe a quick example in the commit message for what the more complicated case will look like is useful.

[the remark this answers is not in the document]
Comment (berfarah) [reply graphql/schemabuilder/types.go]: Added

**Pull request merged**